hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OzoneManagerProtocolClientSideTranslatorPB.java

```diff
@@ -370,13 +396,16 @@ private OMResponse submitRequest(OMRequest omRequest)
 
       return omResponse;
     } catch (ServiceException e) {
-//      throw ProtobufHelper.getRemoteException(e);
       NotLeaderException notLeaderException = getNotLeaderException(e);
       if (notLeaderException == null) {
         throw ProtobufHelper.getRemoteException(e);
```

Comment:
If notLeaderException is null, we will always throw exception without checking the other exception types.

Reply:
Thats the intended behavior right?
We check for NotLeaderException so as to update the error message to indicate why the failure occurred more accurately. For other scenarios, we want to throw the error to the client.

Reply:
Got info from @hanishakoneru , this is done to change the exception message when the exception is type of NotLeaderException.